exovetter/viz_transits.py

```diff
@@ -0,0 +1,148 @@
+# Code to plot and evaluate individual transits.
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy.convolution import convolve, Box1DKernel
+from exovetter import utils
+
+
+def plot_all_transits(time, flux, period, epoch, dur, depth, max_transits=20,
+                      transit_only=False, plot=True):
+    """
+
+    Parameters
+    ----------
+    time : numpy array
+        times of measurements
+    flux : numpy array
+        brightness changes
+    period : float
+        period in same units as time
+    epoch : float
+        epoch of transits in same units and offset as time.
+    dur : float
+        duration of the transit in same units as the time.
+    depth : float
+        depth of possible transit, used to spread out each light curve.
+    max_transits : integer, optional
+        maximum number of transits to plot. The default is 10.
+
+    Returns
+    -------
+    n_has_data = int
+        Number of transits with data in transit (3*duration)
+
+    """
+
+    phases = utils.compute_phases(time, period, epoch, offset=0.25)
+    intransit = utils.mark_transit_cadences(time, period, epoch, dur,
+                                            num_durations=3, flags=None)
+
+    xmin = 0
+    xmax = np.max(phases) * period
+    figwid = 8
+    if transit_only:
+        xmin = np.min(phases[intransit]) * period
+        xmax = np.max(phases[intransit]) * period
+
+        if (xmax - xmin) > (0.5 * period):
+            xmin = (0.25 * period) - 1.25 * dur
+            xmax = (0.25 * period) + 1.25 * dur
+
+        figwid = 4
+
+    offset = 0.25
+    ntransit = np.floor((time - epoch + (offset * period)) / period)
+
+    n_has_data = len(np.unique(ntransit[intransit]))
```

Comment:
Is it worth insisting on a minimum number of points in transit before we say it has data?